aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1123,16 +1137,54 @@ impl AptosVM {
                     &mut new_published_modules_loaded,
                     &storage_gas_params.change_set_configs,
                 ),
-            TransactionPayload::Multisig(payload) => self.execute_multisig_transaction(
-                resolver,
-                session,
-                gas_meter,
-                &txn_data,
-                payload,
-                log_context,
-                &mut new_published_modules_loaded,
-                &storage_gas_params.change_set_configs,
-            ),
+            TransactionPayload::Multisig(payload) => {
+                if !self.is_simulation {
+                    self.execute_multisig_transaction(
+                        resolver,
+                        session,
+                        gas_meter,
+                        &txn_data,
+                        payload,
+                        log_context,
+                        &mut new_published_modules_loaded,
+                        &storage_gas_params.change_set_configs,
+                    )
+                // Otherwise this is simulation.
+                // TODO: de-duplicate with 'execute_multisig_transaction'.
+                } else if let Some(multisig_payload) = payload.transaction_payload.clone() {
+                    match multisig_payload {
+                        MultisigTransactionPayload::EntryFunction(entry_function) => {
+                            aptos_try!({
+                                return_on_failure!(self.execute_multisig_entry_function(
+                                    &mut session,
+                                    gas_meter,
+                                    payload.multisig_address,
+                                    &entry_function,
+                                    &mut new_published_modules_loaded,
+                                ));
+                                let respawned_session = self
+                                    .charge_change_set_and_respawn_session(
+                                        session,
+                                        resolver,
+                                        gas_meter,
+                                        &storage_gas_params.change_set_configs,
+                                        &txn_data,
+                                    )?;
+
+                                self.success_transaction_cleanup(
+                                    respawned_session,
+                                    gas_meter,
+                                    &txn_data,
+                                    log_context,
+                                    &storage_gas_params.change_set_configs,
+                                )
+                            })
+                        },
+                    }
+                } else {
+                    Err(VMStatus::error(StatusCode::MISSING_DATA, None))
```

Comment:
So when can `payload.transaction_payload` be None?

I feel like MISSING_DATA is being misused here. This status code was meant to indicate one particular kind of execution error -- trying to load a non-existent Move resource from the storage.

With that being said I'm aware that this wasn't introduced by this refactoring.

Reply:
What do you think would be a better status here?